### PR TITLE
chore: update deps

### PR DESCRIPTION
**Pull request by ggobbe**
Update dependencies:

- reflect-metadata
- @nestjs/*

### CODE REVIEW

File: packages/nestjs-retry/package.json [hunk 1]
```diff
@@ -60,8 +60,8 @@
     "@geersch/retry": "workspace:^"
   },
   "peerDependencies": {
-    "@nestjs/common": "^9.0.0 || ^10.0.4",
-    "@nestjs/core": "^9.0.0 || ^10.0.4",
-    "reflect-metadata": "^0.1.12"
+    "@nestjs/common": "^10.0.0",
```
Comment (geersch) [inline packages/nestjs-retry/package.json]: There's a bug in NestJS 10 which affects 10.0.3 and lower. That's why the minimum version was 10.0.4.

It works fine with NestJS >= 9.x.x and >= 10.0.4 so you can leave those peer dep requirements.

https://github.com/nestjs/nest/issues/11923

File: packages/nestjs-retry/package.json [hunk 2]
```diff
@@ -60,8 +60,8 @@
...
+    "@nestjs/core": "^10.0.0",
+    "reflect-metadata": "^0.2.1"
```
Comment (geersch) [inline packages/nestjs-retry/package.json]: Could be 

```
"reflect-metadata": "^0.1.12 || ^0.2.0"
```

To allow for >= `0.1.12` or >= `0.2.0`. With a `0.y.z` release every minor version bump is considered breaking.

File: packages/nestjs-retry/package.json (hunk 1 above)
Comment (ggobbe) [reply packages/nestjs-retry/package.json]: right, makes sense 👍

File: packages/nestjs-retry/package.json (hunk 2 above)
Comment (ggobbe) [reply packages/nestjs-retry/package.json]: Good shout 👍

**Pull request merged**